Sources/MongoSwift/ChangeStream.swift

```diff
@@ -0,0 +1,174 @@
+
+  /// A reference to the `mongoc_change_stream_t` pointer.
+  private var changeStream: OpaquePointer?
+
+  /// Used for storing Swift errors.
+  private var swiftError: Error?
+
+  /// The error that occurred while iterating the ChangeStream cursor,
+  /// if one exists. This should be used to check for errors after
+  /// `next()` returns `nil`.
+  public var error: Error? {
+    if let err = self.swiftError {
+        return err
+    }
+
+    var replyPtr = UnsafeMutablePointer<BSONPointer?>.allocate(capacity: 1)
+    defer {
+        replyPtr.deinitialize(count: 1)
+        replyPtr.deallocate()
+    }
+
+    var error = bson_error_t()
+    guard mongoc_cursor_error_document(self.changeStream, &error, replyPtr) else {
+        return nil
+    }
+
+    // If a reply is present, it implies the error occurred on the server. This *should* always be a commandError,
+    // but we will still parse the mongoc error to cover all cases.
+    if let docPtr = replyPtr.pointee {
+        // we have to copy because libmongoc owns the pointer.
+        let reply = Document(copying: docPtr)
+        return extractMongoError(error: error, reply: reply)
+    }
+
+    // Otherwise, the only feasible error is that the user tried to advance a dead change stream cursor,
+    // which is a logic error. We will still parse the mongoc error to cover all cases.
+    return extractMongoError(error: error)
+  }
+
+  /// Decoder for decoding documents into type `T`.
+  internal let decoder: BSONDecoder
+
+  /// Returns the next `T` in the change stream or nil if there is no next value.
+  /// Will block for a maximum of `maxAwaitTimeMS` milliseconds as specified in the
+  /// `ChangeStreamOptions`, or for the server default timeout if omitted.
+  public func next() -> T? {
+    let cursor = self.changeStream
+    // Allocate space for a reference to a BSON pointer.
+    let out = UnsafeMutablePointer<BSONPointer?>.allocate(capacity: 1)
+
+    // Check that there’s another document in the stream.
+    guard mongoc_change_stream_next(cursor, out) else {
+      return nil
+    }
+
+    defer {
+        out.deinitialize(count: 1)
+        out.deallocate()
+    }
+
+    guard let pointee = out.pointee else {
+        fatalError("mongoc_change_stream_next returned true, but document is nil")
+    }
+
+    // we have to copy because libmongoc owns the pointer.
+    let doc = Document(copying: pointee)
+
+    // Update the resumeToken with the `_id` field from the document.
+    if let resumeToken = doc["_id"] as? Document {
+        self.resumeToken = ChangeStreamToken(resumeToken: resumeToken)
+    } else {
+        self.swiftError = RuntimeError.internalError(message: "_id field is missing from the change stream document.")
+        return nil
+    }
+
+    do {
+      return try decoder.decode(T.self, from: doc)
+    } catch {
+      self.swiftError = error
+    }
+
+    return nil
+  }
+
+  /**
+   * Returns the next `T` in this change stream or
```

Comment:
These docstrings can be expanded to the full width that our formatter accepts (I think its 100? maybe 120? characters)